tests/manage/pv_services/pv_encryption/test_pvc_with_tc_overrie_vn.py

```diff
@@ -0,0 +1,116 @@
+if config.ENV_DATA.get("vault_hcp"):
+    argvalues = [
+        pytest.param(
+            "v1", kmsprovider, True, marks=pytest.mark.polarion_id("OCS-4639")
+        ),
+        pytest.param(
+            "v2", kmsprovider, True, marks=pytest.mark.polarion_id("OCS-4641")
+        ),
+    ]
+else:
+    argvalues = [
+        pytest.param(
+            "v1", kmsprovider, False, marks=pytest.mark.polarion_id("OCS-4638")
+        ),
+        pytest.param(
+            "v2", kmsprovider, False, marks=pytest.mark.polarion_id("OCS-4640")
+        ),
+    ]
+
+
+@pytest.mark.parametrize(
+    argnames=argnames,
+    argvalues=argvalues,
+)
+@tier1
+@skipif_ocs_version("<4.8")
+class TestEncryptedRbdBlockPvcSnapshot(ManageTest):
+    """
+    Tests to check Tenant configmap override vault namespace or not
+    1.
+
+    """
+
+    @pytest.fixture(autouse=True)
+    def setup(
+        self,
+        kv_version,
+        kms_provider,
+        use_vault_namespace,
+        pv_encryption_kms_setup_factory,
+        project_factory,
+        storageclass_factory,
+        pvc_factory_class,
+    ):
+        """
+        Setup csi-kms-connection-details configmap
+
+        """
+
+        log.info("Setting up csi-kms-connection-details configmap")
+        self.kms = pv_encryption_kms_setup_factory(kv_version, use_vault_namespace)
+        log.info("csi-kms-connection-details setup successful")
+
+        # Create a project
+        self.proj_obj = project_factory()
+
+        # Create an encryption enabled storage class for RBD
+        self.sc_obj = storageclass_factory(
+            interface=constants.CEPHBLOCKPOOL,
+            encrypted=True,
+            encryption_kms_id=self.kms.kmsid,
+        )
+
+    def test_status_with_tenant_configmap(self, multi_pvc_factory, kv_version):
+
+        # Create new backend-path
+        vault_resource_name = create_unique_resource_name("test", "vault")
+        self.kms.vault_create_backend_path(
+            backend_path=vault_resource_name, kv_version=kv_version
+        )
+
+        # Create policy
+        self.kms.vault_create_policy(policy_name=vault_resource_name)
+
+        # Create a configmap in the tenant namespace to override the vault namespace as shown below:
+        self.kms.create_tenant_configmap(
+            self.proj_obj.namespace, vaultBackendPath=f"{vault_resource_name}"
+        )
+
+        self.kms.create_vault_csi_kms_token(namespace=self.proj_obj.namespace)
+
+        # Create New PVC and check status
+        self.pvc_size = 1
+        self.pvc_obj = multi_pvc_factory(
+            interface=constants.CEPHBLOCKPOOL,
+            project=self.proj_obj,
+            storageclass=self.sc_obj,
+            access_modes=[
+                f"{constants.ACCESS_MODE_RWX}-Block",
+                f"{constants.ACCESS_MODE_RWO}-Block",
+            ],
+            size=self.pvc_size,
+            status=constants.STATUS_BOUND,
+            num_of_pvc=2,
+            wait_each=False,
+        )
```

Comment:
The following steps are missing from the test:
 1. Mounting the PVC in an app pod and running IO
 2. Checking if the encryption key is actually created in the new backend path

Reply:
Added code